teps/0089-nonfalsifiable-provenance-support.md

```diff
@@ -224,24 +309,298 @@ The entrypointer image should also be able to see that this flag is set and acco
+
+- The TektonController and TektonChains would be able to continue verification of TaskRuns for certificate validity time
+- The TaskRun pods would be able to sign if they already obtained their SVIDs
+- No new TaskRun pods would be able to sign when the SPIRE server goes down - as new entries can't be created, and new SVIDs can't be minted
+
+Another failure could be the SPIRE agents (daemonset), which would result in not being able to obtains SVIDs for that particular node while it is down (with the similar caveats for temporary failure and cert lifetime).
+
+## Design Details - Verification policy
+
+When the verifiers are unable to verify a document, either because the hashes don't match up or it is unable to obtain the trust bundle, what should the desirable action be? There are several options here:
+
+- Stop execution of a TaskRun
+- Indicate that the TaskRun is no longer verifiable but continue execution
+
+While this is at an early stage, we opt for indicating that a document is not verifiable, future additions can include ability to configure the action to be taken.
+
+
+## Design Details - Verfication of data
+
+The following details how verification is done with relation to the verification authority (SPIRE server), as well as the materials produced by the signing process. The following are needed in order to perform verification, along with their purpose in the verification:
+
+- Trust Bundle: Verification authority (CA) - provided by SPIRE server
+- Cert (x509): Creates a metadata that a key-pair K belongs to a workload/pod X, and this information is endorsed by the verification authority (CA)
+- Signature: This content was verified, and the evidence was produced by keypair K
+
+The verification process is as follows:
+- Obtain the Trust Bundle independently from the SPIRE server
+- Obtain the x509 cert and the signature from the signed object metadata
+- Verify that the x509 cert is endorsed/signed by the authority in the Trust Bundle
+- Verify that the x509 cert belongs to the right workload, i.e. if signed by the TektonController, the cert should indicate the URI of the TektonController, and the same for each individual TaskRun
+- Verify that the signature was signed by the key as indicated in the x509 cert
+
+As we saw above, when an object is signed, we create a signature, which we accompany with an x509 certificate which links the signature to the signer (in this case, the workload that signs it - whether it be a TaskRun pod or the TektonController). These keys which are used to generate the signature are short-lived keys, and the SPIRE server does not keep track of each individual certificate generated, therefore, the x509 certificate is required to be stored with the signature data.
```

Comment:
It does! I think I'm nit picking at this point so please feel free to disregard this but I think this could be more clearly highlighted earlier in the proposal (again totally my own opinion and very biased by how ive been consuming updates to this TEP, might be totally different for someone reading it from scratch) - im wondering if it would make sense to have more details in the requirements that help explain this and other choices that were made, e.g. something like:

* It must be possible for the controller to verify signatures generated by executing pods without needing to request the pod's public key (which is short lived and may not be available at this point)

Reply:
gotcha yea, ill weave that in!